evalml/utils/nullable_type_utils.py

```diff
@@ -0,0 +1,108 @@
+import woodwork as ww
+
+
+# --> add to __init__.py - figure out circular dependency
+def _downcast_nullable_X(X, handle_boolean_nullable=True, handle_integer_nullable=True):
+    """Removes Pandas nullable integer and nullable boolean dtypes from data by transforming
+        to other dtypes via Woodwork logical type transformations.
+
+    Args:
+        X (pd.DataFrame): Input data of shape [n_samples, n_features] whose nullable types will be changed.
+        handle_boolean_nullable (bool, optional): Whether or not to downcast data with BooleanNullable logical types.
+        handle_integer_nullable (bool, optional): Whether or not to downcast data with IntegerNullable or AgeNullable logical types.
+
+
+    Returns:
+        X with any incompatible nullable types downcasted to compatible equivalents.
+    """
+    # --> consider adding param for expecting there to not be any nans present so we're
+    # notified if we're ever unknowingly converting to Double or Categorical when we shouldnt in automl search
```

Comment:
Just a small behavior consideration that I am not going to implement unless/until a need for it arises. Likely, it'd happen during the integration into AutoMLSearch.